app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettings.kt

```diff
@@ -214,7 +220,10 @@ private fun WallpaperThumbnailItem(
         Modifier
     }
 
-    val bitmap = loadWallpaperResource(wallpaper)
+    var bitmap by remember { mutableStateOf(loadWallpaperResource(wallpaper)) }
+    DisposableEffect(LocalConfiguration.current.orientation) {
+        onDispose { bitmap = loadWallpaperResource(wallpaper) }
+    }
     // Completely avoid drawing the item if a bitmap cannot be loaded and is required
     if (bitmap == null && wallpaper != defaultWallpaper) return
     Surface(
```

Comment:
I would add a new line in between this to visually separate out what I call the 'logical' blocks. Just makes thing easier to read when there are clear separation of code blocks (for example, block for variable init, if/when statements, Compose content).

Reply:
Updated